broker/src/main/java/io/camunda/zeebe/broker/system/management/BrokerAdminServiceImpl.java

```diff
@@ -134,39 +134,24 @@ private CompletableFuture<PartitionStatus> getPartitionStatus(final ZeebePartiti
             partitionStatus.completeExceptionally(error);
             return;
           }
-          if (currentRoleFuture.join() == Role.LEADER) {
-            final var streamProcessor = streamProcessorFuture.join();
-            final var exporterDirector = exporterDirectorFuture.join();
-            if (streamProcessor.isEmpty() || exporterDirector.isEmpty()) {
-              partitionStatus.completeExceptionally(
-                  new IllegalStateException(
-                      "No streamprocessor or exporter found for leader partition."));
-            } else {
-              getLeaderPartitionStatus(
-                  partition, streamProcessor.get(), exporterDirector.get(), partitionStatus);
-            }
-
+          final var role = currentRoleFuture.join();
+          final var streamProcessor = streamProcessorFuture.join();
+          final var exporterDirector = exporterDirectorFuture.join();
+
+          if (streamProcessor.isEmpty() || exporterDirector.isEmpty()) {
+            partitionStatus.completeExceptionally(
+                new IllegalStateException(
+                    "No streamprocessor or exporter found for leader partition."));
           } else {
-            getFollowerPartitionStatus(partition, partitionStatus);
+            getLeaderPartitionStatus(
+                role, partition, streamProcessor.get(), exporterDirector.get(), partitionStatus);
           }
         });
     return partitionStatus;
   }
 
-  private void getFollowerPartitionStatus(
-      final ZeebePartition partition, final CompletableFuture<PartitionStatus> partitionStatus) {
-    final var snapshotId = getSnapshotId(partition);
-    final var processedPositionInSnapshot =
-        snapshotId
-            .flatMap(FileBasedSnapshotId::ofFileName)
-            .map(FileBasedSnapshotId::getProcessedPosition)
-            .orElse(null);
-    final var status =
-        PartitionStatus.ofFollower(snapshotId.orElse(null), processedPositionInSnapshot);
-    partitionStatus.complete(status);
-  }
-
   private void getLeaderPartitionStatus(
```

Comment:
:x: Should be renamed to `getLeaderPartitionStatus` as it is no longer specifically for partition leaders.